ethereum/jsonrpc/src/main/java/tech/pegasys/pantheon/ethereum/jsonrpc/internal/results/DebugTraceBlockResult.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright 2019 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+package tech.pegasys.pantheon.ethereum.jsonrpc.internal.results;
+
+import tech.pegasys.pantheon.ethereum.debug.TraceFrame;
+import tech.pegasys.pantheon.ethereum.jsonrpc.internal.processor.BlockTrace;
+import tech.pegasys.pantheon.ethereum.jsonrpc.internal.processor.TransactionTrace;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+import com.fasterxml.jackson.annotation.JsonGetter;
+import com.fasterxml.jackson.annotation.JsonPropertyOrder;
+
+@JsonPropertyOrder({"gas", "structLogs"})
+public class DebugTraceBlockResult {
```

Comment:
It's missing `value` property (it's listed in issue), any hints how block's value can be computed ?

Reply:
The issue description is missing an important element here unforunately - the result should be an array of transaction trace results.  So:

```
[
{
  gas: 999,
  returnValue: "",
  structLogs: [{
      depth: 1,
      error: "",
      gas: 99999,
      gasCost: 2,
      memory: null,
      op: "PUSH1",
      pc: 0,
      stack: [],
      storage: {}
  },
  ... (more transactions)
]
```

The `returnValue` is then the return value of the transaction the same way `DebugTraceTransactionResult` extracts it.  In fact, I suspect `DebugTraceBlockResult` isn't required and the success response is just `return new JsonRpcSuccessResponse(request.getId(), listOfDebugTraceTransactionResult)`

Reply:
Thanks, i'll change it.